packages/babel-helper-define-polyfill-provider/src/types.ts

```diff
@@ -0,0 +1,120 @@
+import type { NodePath } from "@babel/traverse";
+import { types as t } from "@babel/core";
+import createMetaResolver from "./meta-resolver";
+
+type ObjectMap<T> = {
+  [k: string]: T;
+};
+
+export type Pattern = string | RegExp;
+
+export type MissingDependenciesOption =
+  | false
+  | {
+      log?: "per-file" | "deferred";
+      inject?: "import" | "throw";
+      // When true, log all the polyfills without checking if they are installed
+      all?: boolean;
+    };
+
+export type ProviderOptions<Opts = {}> = {
+  include?: Pattern[];
+  exclude?: Pattern[];
+} & Opts;
+
+export type PluginOptions = {
+  method: MethodString;
+  targets?: {
+    browsers: string | string[];
+    [target: string]: string | number;
+  };
+  ignoreBrowserslistConfig?: boolean;
+  configPath?: string;
+  debug?: boolean;
+  include?: Pattern[];
+  exclude?: Pattern[];
+  shouldInjectPolyfill?: (name: string, shouldInject: boolean) => boolean;
+  missingDependencies?: MissingDependenciesOption;
+  absoluteImports?: boolean | string;
+};
+
+export type PolyfillProvider<Opts = {}> = (
+  api: ProviderApi,
+  options: ProviderOptions<Opts>,
+  dirname: string,
+) => ProviderResult;
+
+export type PolyfillProviderInternal<Opts> = PolyfillProvider<Opts>;
```

Comment:
this was previously `export type PolyfillProviderInternal = PolyfillProvider<Opts>;` which gave a bunch of TS errors but somehow was valid Flow? afaict, should be a generic on both sides